com.ibm.wala.core/src/com/ibm/wala/ipa/callgraph/propagation/InstanceFieldKey.java

```diff
@@ -18,6 +18,7 @@
  */
 public class InstanceFieldKey extends AbstractFieldPointerKey {
   private final IField field;
+  private Integer hashCode = null;
```

Comment:
I suspect that such caching could be added in lots of places.  Do you have enough profiling info to see if it's worth it elsewhere?